src/components/ModularTable/ModularTable.tsx

```diff
@@ -25,21 +25,40 @@ export type Props<D extends Record<string, unknown>> = {
    * Optional extra row to display underneath the main table content.
    */
   footer?: ReactNode;
+  getTableProps?: TablePropGetter<D>;
+  getHeaderProps?: (column: Column<D>) => Record<string, unknown>;
```

Comment:
It seems a shame that there isn't a more specific generic or return type for these that make them a bit more obvious how they're supported to be used (or at least at a quick look at react-table I couldn't see anything more specific).

On second thoughts, should/could we limit these to the element props they're used with e.g. `...=> HTMLProps<HTMLTableRowElement>` or do we need to allow for overriding specific react-table props as well?